### ISSUE TITLE
Swagger isn't obeying required=True for IPv4 field

### ISSUE THREAD

**Issue by cjproud** [marshmallow-code/flask-smorest#623]
I have a very simple schema which is used to deserialize and validate an IP address from a query string, such as `URL/?ip=X.X.X.X`:

```python
class NFSIPQueryArgsSchema(Schema):
    ip = fields.IPv4(required=True)
```

however even though the `required=True` field is set, attempting to query the endpoint via Swagger is still allowed even with an empty IP field:

<img width="1423" alt="Screenshot 2024-03-14 at 4 06 04 pm" src="https://github.com/marshmallow-code/flask-smorest/assets/41252312/148f0e20-8386-4805-8ecd-a6312fe283cc">

if I change this field to a string, rather than an IP address:

```python
class NFSIPQueryArgsSchema(Schema):
    ip = fields.String(required=True)
```

Swagger won't let me query the endpoint as expected:

<img width="1404" alt="Screenshot 2024-03-14 at 4 05 47 pm" src="https://github.com/marshmallow-code/flask-smorest/assets/41252312/5c628bb2-adad-41af-825f-13f9731293a7">

I'm using flask-smorest v0.42.2 and flask 3.0.1.

**Comment by lafrech** [comment]
Not sure what happens, here. The field is displayed as required. 

The difference I see is that the IP field is not a "string" field. We may want to improve this.

As per https://swagger.io/docs/specification/data-models/data-types/, we could define the type as `string` and the format as `ipv4`.

This should be an easy contribution. It happens here in apispec: https://github.com/marshmallow-code/apispec/blob/dev/src/apispec/ext/marshmallow/field_converter.py#L23. Let's add `ipv4`, `ipv6` and `ip` if for the `IP` field.

Would you like to take a stab at it?

**Comment by cjproud** [comment]
Sure thing - I'm about to knock off for the day but can give it a stab tomorrow.

Quick question (and definitely a dumb one), I haven't done development on a package dependency before...if I clone `marshmallow-code/apispec` to my machine how do I tell `flask-smorest` to use that as a dependency rather than the version from PyPI?

**Comment by cjproud** [comment]
I should clarify, I want to do the above to test that my changes work 👍

**Comment by lafrech** [comment]
Thanks for this.

In your flask-smorest virtualenv, install your modified apispec with `pip install -e`.

**Comment by cjproud** [comment]
Hey @lafrech I've just made a PR in `apispec`: https://github.com/marshmallow-code/apispec/pull/892. I hope that's what you were looking for :)

**Comment by lafrech** [comment]
Fixed in apispec 6.6.0.

Thanks.
